### PR TITLE
Add quality declaration documents

### PR DESCRIPTION
**Pull request by gbiggs**
This PR adds [quality declaration documents](https://www.ros.org/reps/rep-2004.html) and (where necessary) basic README files too all packages in `rmf_demos`.

The resource packages meet the requirements to be Quality Level 3. All source-code-containing packages are currently QL4.

### CODE REVIEW

File: rmf_gazebo_plugins/QUALITY_DECLARATION.md [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+`rmf_robot_sim_gazebo_plugins` follows the recommended guidelines for ROS Core packages in the [ROS 2 Developer Guide](https://index.ros.org/doc/ros2/Contributing/Developer-Guide/#package-requirements).
+
+### Change Requests [2.i]
+
+`rmf_robot_sim_gazebo_plugins` requires that all changes occur through a pull request.
+
+### Contributor Origin [2.ii]
+
+`rmf_robot_sim_gazebo_plugins` does not require a confirmation of contributor origin.
```
Comment (luca-della-vedova) [inline rmf_gazebo_plugins/QUALITY_DECLARATION.md:46]: I guess this changed now?

File: rmf_gazebo_plugins/QUALITY_DECLARATION.md [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+
+### Peer Review Policy [2.iii]
+
+All pull requests must have at least 1 peer review.
+
+### Continuous Integration [2.iv]
+
+All pull requests must pass CI on all platforms supported by RMF.
+The CI checks only that the package builds.
+The most recent CI results can be seen on [the workflow page](https://github.com/open-rmf/rmf_simulation/actions).
```
Comment (luca-della-vedova) [inline rmf_gazebo_plugins/QUALITY_DECLARATION.md:56]: Still TODO, will link PR when this is done

File: rmf_gazebo_plugins/QUALITY_DECLARATION.md [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+### Documentation Policy [2.v]
+
+All pull requests must resolve related documentation changes before merging.
+
+## Documentation [3]
+
+### Feature Documentation [3.i]
+
+`rmf_robot_sim_gazebo_plugins` is documented in the [parent repository's README.md file](https://github.com/open-rmf/rmf_simulation/blob/master/README.md).
```
Comment (luca-della-vedova) [inline rmf_gazebo_plugins/QUALITY_DECLARATION.md:66]: TODO, same as above

File: rmf_gazebo_plugins/QUALITY_DECLARATION.md [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+
+`rmf_robot_sim_gazebo_plugins` does not have a public API.
+
+### License [3.iii]
+
+The license for `rmf_robot_sim_gazebo_plugins` is Apache 2.0, the type is declared in the [package.xml](package.xml) manifest file, and a full copy of the license is in the repository level [LICENSE](../LICENSE) file.
+
+### Copyright Statement [3.iv]
+
+The copyright holders each provide a statement of copyright in each source code file in `rmf_demo_tasks`.
```
Comment (luca-della-vedova) [inline rmf_gazebo_plugins/QUALITY_DECLARATION.md:78]: I guess leftover from copying from another repo?

File: rmf_gazebo_plugins/QUALITY_DECLARATION.md [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+
+### Quality declaration document [3.v]
+
+This quality declaration is linked in the [README file](README.md).
```
Comment (luca-della-vedova) [inline rmf_gazebo_plugins/QUALITY_DECLARATION.md:82]: TODO

File: rmf_gazebo_plugins/QUALITY_DECLARATION.md [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+
+#### eigen
+
+`eigen` does not declare a quality level.
+It is assumed to be QL1 based on wide-spread use.
+
+## Platform Support [6]
+
+`rmf_robot_sim_gazebo_plugins` does not support all of the tier 1 platforms as described in [REP-2000](https://www.ros.org/reps/rep-2000.html#support-tiers).
+`rmf_robot_sim_gazebo_plugins` supports ROS Eloquent and ROS Foxy.
```
Comment (luca-della-vedova) [inline rmf_gazebo_plugins/QUALITY_DECLARATION.md:163]: Support for Eloquent has been dropped

**Review by luca-della-vedova — CHANGES_REQUESTED** [review]
Note, the files are now in the wrong folders (i.e. `rmf_gazebo_plugins`). I just put the comments for one file but they usually apply to all files (i.e. missing github actions, DCO / GPG signing being enforced, typos due to copying the declaration from another repo)

File: rmf_gazebo_plugins/QUALITY_DECLARATION.md (hunk 2 above)
Comment (luca-della-vedova) [reply rmf_gazebo_plugins/QUALITY_DECLARATION.md:56]: Done in #11

File: rmf_gazebo_plugins/QUALITY_DECLARATION.md (hunk 5 above)
Comment (mxgrey) [reply rmf_gazebo_plugins/QUALITY_DECLARATION.md:83]: The package-specific README.md does link to the quality declaration.

**Comment by mxgrey** [comment]
I just noticed that the packages these quality declarations are targeting no longer exist as they were split up between a bunch of new packages. I'm working on refactoring the quality declarations into the new packages.

**Review by mxgrey — APPROVED** [review]

**Review by mxgrey — APPROVED** [review]

**Comment by mxgrey** [comment]
@luca-della-vedova, it seems that GitHub will need you to change your review to an approval before it's willing to merge.

**Review by luca-della-vedova — APPROVED** [review]
Let's do it! :tada:

**Pull request merged**